### PR TITLE
Add Kubernetes on AWS EKS Hello World Tutorial

### PR DESCRIPTION
**Pull request by metral**

### CODE REVIEW

**Comment by metral** [comment]
The corresponding code for the example linked in the tutorial: https://github.com/pulumi/pulumi-eks/pull/61

PTAL at both as the linked PR is a dependency of this tutorial, and needs to be merged first.

File: quickstart/aws/tutorial-eks.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    });
+
+    // Export the clusters' kubeconfig.
+    export const kubeconfig = cluster.kubeconfig
+    ```
+
+1. Add the required dependencies:
+
+	```bash
+	$ yarn add @pulumi/eks @pulumi/aws-infra
```
Comment (joeduffy) [inline quickstart/aws/tutorial-eks.md]: I _think_ we tend to prefer `npm` in these tutorials. So

```
$ npm install --save @pulumi/eks @pulumi/aws-infra
```

File: quickstart/aws/tutorial-eks.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+	```
+
+1.  To preview and deploy changes, run `pulumi update`. The command shows a preview of the resources that will be created and prompts on whether to proceed with the deployment.  Note that the stack itself is counted as a resource, though it does not correspond to a physical cloud resource.
+
+    ```bash
+	$ pulumi update
```
Comment (joeduffy) [inline quickstart/aws/tutorial-eks.md]: Nit, tabs versus spaces?

File: quickstart/aws/tutorial-eks.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+	+      ├─ aws:ec2:RouteTableAssociation           vpc-nat-0                         	create
+	+      ├─ aws:ec2:RouteTable                      vpc-nat-1                         	create
+	+      ├─ aws:ec2:RouteTable                      vpc-nat-0                         	create
+	+      ├─ aws:ec2:RouteTableAssociation           vpc-1                             	create
+	+      └─ aws:ec2:RouteTableAssociation           vpc-0                             	create
+
+	Resources:
+		+ 42 to create
+
+	clusterng (eks-demo):
```
Comment (joeduffy) [inline quickstart/aws/tutorial-eks.md]: Maybe make a note here that you should select "yes" to confirm the deployment. I'd probably just copy whatever we do in other tutorials.

**Comment by joeduffy** [comment]
Hmm, having a tutorial in addition to the example is highly valuable ... Why not go forward with it?

**Comment by metral** [comment]
I could add the proposed example in that same `pulumi/examples/aws-ts-eks` dir as the existing one, but I wasn't sure if that'd be unnecessarily extraneous, not to mention writing a tutorial for it.

I can still proceed forward, just lmk what makes the most sense.

**Comment by joeduffy** [comment]
Having a tutorial on the docs website tends to significantly increase its visibility and utility. I don't think we have one for EKS currently -- it's a shame not to publish the excellent one you've written.

I'm on the fence on the question of the examples repo, however, I do think showing how to deploy K8s objects into a newly provisioned cluster is sufficiently subtle and interesting that it's worth having.

**Comment by lukehoban** [comment]
Yes - let's definitely add this tutorial to the docs site.  

And totally fine having both `aws-ts-eks` and `aws-ts-eks-hello-world` in the examples repo.

File: quickstart/aws/tutorial-eks.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        {
+            provider: cluster.provider,
+        }
+    );
+    export const deploymentName = deployment.metadata.apply(m => m.name);
+    ```
+
+1.  Run `pulumi update` to preview and deploy the changes. You'll see two changes: the `Namespace` in Kubernetes to deploy into, and the `Deployment` resource for the new NGINX app. Pulumi understands which changes to a given cloud resource can be made in-place, and which require replacement, and computes the minimally disruptive change to achieve the desired state.
+
+    As part of the update, you'll see the two new objects in the output:
```
[the remark this answers is not in the document]
Comment (metral) [reply quickstart/aws/tutorial-eks.md]: Great point. I've included the wording @lukehoban[ used in the community slack](https://pulumi-community.slack.com/archives/C84L4E3N1/p1545597765076500?thread_ts=1545559366.075500&cid=C84L4E3N1) for reference.

**Comment by metral** [comment]
I've reopened the PR for this EKS tutorial, and addressed all feedback. 

I've also created a new `aws-ts-eks-hello-world` example in `pulumi/examples` that this tutorial depends on for its link to source code that will have to be merged in first: https://github.com/pulumi/examples/pull/219

PTAL

**Pull request merged**